pkg/templates/templates.go

```diff
@@ -160,7 +173,7 @@ func (t *TemplateResolver) ResolveTemplate(tmplMap interface{}) (interface{}, er
 	}
 
 	var buf strings.Builder
-	err = tmpl.Execute(&buf, "")
```

Comment:
When I originally saw this line I thought "oooh, can users put other stuff into the template?". Is `.ClusterName` the only thing they might want? If not, should the config just take a whole context map, and use it here?

I understand keeping the config smaller, and not allowing arbitrary things. I'm just curious if this was brought up already.

Reply:
It's a good point. @ckandag, what do you think?

The benefit is that the caller has full control of the context that is available during template execution. This would also not require us to update `go-template-utils` every time we want to add a new field.

The downside is that `.ClusterName` is cleaner syntax and what has been approved as opposed to `.["clusterName"]`.

Reply:
I would think,  Ideally util package provides functionality thats common across all consumers of the pkg while any consumer specific details  like the  context or rbac etc are handled on the consumer side.   To the util pkg , the input and output  is just some valid yaml string , the contents in it - what  Kubernetes CR - shouldn't matter at all.


But we definitely need the syntax  to be  .ClusterName,    if there is a way to achieve that while allowing the consumer to pass in the context   without hardcoded it in util pkg ( via reflection or such)  that would be nice.  If not, we can live with this for now.

Also , as a consumer e.g. config-policyc-controller   the kubeclient, delimiters are one time settings that are generally common through out all  invocations of the  ResolveTemplate  call.   However, the context  is different for each invocation , just as  the  template yaml string  or the result of calling fromSecret or lookup could be different.

I really dont think the context should be part of config but should be passed optionally as param to ResolveTemplate  function.

Reply:
Let me have a go with reflection and see if I can come up with a way to do this.